src/Server/Resource/DoctrineResource.php

```diff
@@ -543,20 +548,21 @@ public function patch($id, $data)
         }
             // @codeCoverageIgnoreEnd
 
-        // Hydrate entity with patched data
-        $this->getHydrator()->hydrate((array) $data, $entity);
```

Comment:
Not sure I am OK with this.
In some listeners I want the already updated entity.
Asume I want to change a field in the PRE_UPDATE. It will be overridden with the $data afterwards. Also: I don't have access to the $data in the listener.

Maybe it's a good idea to still do the hydration, but add the previous results as an extra parameter?
Something like this for ODM:
```php
$class = $this->getObjectManager()->getClassMetadata(get_class($entity));
$uow = $this->getObjectManager()->getUnitOfWork();
$originalData = $uow->getOriginalDocumentData($entity);
```

Just some thought :)